plugins/filter_kubernetes/kube_meta.c

```diff
@@ -441,6 +452,15 @@ static void extract_container_hash(struct flb_kube_meta *meta,
                             container_hash_len = tmp_len;
                         }
                     }
+                    else if (k2.via.str.size == sizeof("image") - 1 &&
+                              !strncmp(k2.via.str.ptr,
+                                       "image",
+                                       k2.via.str.size)) {
+                        if (extract_hash(v2.ptr, v2.size, &tmp, &tmp_len) == 0) {
```

Comment:
extract_hash() is not going to work as expected here. It will return only last part after ":" .
For example if image is "quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.24.1" only "0.24.1" is going to be returned. extract_hash() is meant to be only for hash extraction out of container/image IDs.
I would just do:

```
container_image = v2.ptr;
container_image_len = v2.size;
```

Reply:
I see, so extract_hash is needed to throw away 'docker-pullable://' and 'docker://. Will fix